pkgs/development/libraries/libsoup/default.nix

```diff
@@ -1,33 +1,41 @@
 { stdenv, fetchurl, glib, libxml2, pkgconfig
-, gnomeSupport ? false, libgnome_keyring, sqlite, glib_networking, gobjectIntrospection
+, gnomeSupport ? true, libgnome_keyring, sqlite, glib_networking, gobjectIntrospection
+, valaSupport ? true, vala
 , libintlOrEmpty
 , intltool, python }:
 let
-  majorVersion = "2.52";
-  version = "${majorVersion}.2";
+  majorVersion = "2.54";
+  version = "${majorVersion}.1";
 in
 stdenv.mkDerivation {
   name = "libsoup-${version}";
 
   src = fetchurl {
     url = "mirror://gnome/sources/libsoup/${majorVersion}/libsoup-${version}.tar.xz";
-    sha256 = "1p4k40y2gikr6m8p3hm0vswdzj2pj133dckipd2jk5bxbj5n4mfv";
+    sha256 = "0cyn5pq4xl1gb8413h2p4d5wrn558dc054zhwmk4swrl40ijrd27";
   };
 
-  patchPhase = ''
+  prePatch = ''
     patchShebangs libsoup/
+  '' + stdenv.lib.optionalString valaSupport
+  ''
+     substituteInPlace libsoup/Makefile.in --replace "\$(DESTDIR)\$(vapidir)" "\$(DESTDIR)\$(girdir)/../vala/vapi"
   '';
 
   outputs = [ "dev" "out" ];
 
-  buildInputs = libintlOrEmpty ++ [ intltool python sqlite ];
+  buildInputs = libintlOrEmpty ++ [ intltool python sqlite ]
+    ++ stdenv.lib.optionals valaSupport [ vala ];
   nativeBuildInputs = [ pkgconfig ];
   propagatedBuildInputs = [ glib libxml2 gobjectIntrospection ]
     ++ stdenv.lib.optionals gnomeSupport [ libgnome_keyring ];
   passthru.propagatedUserEnvPackages = [ glib_networking.out ];
 
   # glib_networking is a runtime dependency, not a compile-time dependency
-  configureFlags = "--disable-tls-check --enable-vala=no" + stdenv.lib.optionalString (!gnomeSupport) " --without-gnome";
+  configureFlags = "--disable-tls-check"
+    + stdenv.lib.optionalString (!valaSupport) " --enable-vala=no"
```

Comment:
I used to have gnomeSupport over there for testing purposes.
Feel free to commit over it.

Personally, I don't care either way :)